python3/python3.bzl

```diff
@@ -19,8 +19,8 @@
 # git repository.
 
 DIGESTS = {
-    # "gcr.io/distroless/python3:debug" circa 2019-10-11 13:46 -0400
-    "debug": "sha256:44b13e14537accc5784b475bd756bba289135d4554e204d204f1455eb0c0d7da",
-    # "gcr.io/distroless/python3:latest" circa 2019-10-11 13:46 -0400
-    "latest": "sha256:de110a5eb0edb950dc4653ae0288f530b8c2af77e44d9bebab117ed0b74d5426",
+    # "gcr.io/distroless/python3:debug" circa 2020-10-19 11:55 -0700
+    "debug": "sha256:1d992aa6631cacd44746d5d2915b21ad93869cfb2519f90bd516a244b528b6fe",
```

Comment:
It looks like one of these is problematic. Try manually reverting the changes in this file?

Reply:
Hmm. I reran the script completely and it now is reporting errors for both python and go containers :).The errors is non-deterministic, based on which CI step you look at -- which means it may also have messed up any others, but Bazel is reporting the first error.

Locally, I also get errors for `cc_image`:
```
ERROR: /Users/dan/rules_docker/tests/container/rust/BUILD:20:11: //tests/container/rust:rust_image depends on @cc_image_base//image:image in repository @cc_image_base which failed to fetch. no such package '@cc_image_base//image': SHA256 of the image specified does not match SHA256 of the pulled image. Expected sha256:ac75dbf0b249e5e3758134458d447bfd6a85d74c262ecb7cd88df68a3f53c6e3, but pulled image with sha256:c4014bdecaede16f767f8cfc496f968736bc08632bf54a37c004820e85cd8209. It is possible that you have a pin to a manifest list which points to another image, if so, change the pin to point at the actual Docker image
```

Do you know what the pin it's talking about refers to?

Reply:
Playing around:

```
$ ~/go/bin/crane digest gcr.io/distroless/cc:latest 
2020/10/30 13:17:44 No matching credentials were found, falling back on anonymous
sha256:ac75dbf0b249e5e3758134458d447bfd6a85d74c262ecb7cd88df68a3f53c6e3

$ ~/go/bin/crane manifest gcr.io/distroless/cc_:latest     
2020/10/30 13:15:32 No matching credentials were found, falling back on anonymous
2020/10/30 13:15:32 fetching manifest gcr.io/distroless/cc_:latest: GET https://gcr.io/v2/distroless/cc_/manifests/latest: MANIFEST_UNKNOWN: Failed to fetch "latest" from request "/v2/distroless/cc_/manifests/latest".
dan@dan ~ % ~/go/bin/crane manifest gcr.io/distroless/cc:latest 
2020/10/30 13:15:34 No matching credentials were found, falling back on anonymous
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 947,
         "digest": "sha256:c4014bdecaede16f767f8cfc496f968736bc08632bf54a37c004820e85cd8209",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 947,
         "digest": "sha256:ec7bfcbba262708761909c65c9d26358a1785b849bf32507fa785ddaed77e9bd",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      }
   ]
}%
```

So I can see where both the shas in the error are coming from.

Reply:
I'm guessing that https://github.com/google/go-containerregistry/issues/625 is related.

Reply:
Ah, and duh, I have confirmed that the sha256sum of the manifest is the string returned by `crane digest`:

```
$  ~/go/bin/crane manifest gcr.io/distroless/cc:latest | sha256sum 
2020/10/30 13:44:37 No matching credentials were found, falling back on anonymous
ac75dbf0b249e5e3758134458d447bfd6a85d74c262ecb7cd88df68a3f53c6e3  -
```